onebusaway-api-core/src/main/java/org/onebusaway/api/model/transit/BeanFactoryV2.java

```diff
@@ -159,12 +195,49 @@ public EntryWithReferencesBean<BlockV2Bean> getBlockResponse(BlockBean block) {
 
   public EntryWithReferencesBean<StopWithArrivalsAndDeparturesV2Bean> getResponse(
       StopWithArrivalsAndDeparturesBean result) {
-    return entry(getStopWithArrivalAndDepartures(result));
+    EntryWithReferencesBean<StopWithArrivalsAndDeparturesV2Bean> response = entry(getStopWithArrivalAndDepartures(result));
+
+    String agencyId = response
+            .getEntry()
+            .getArrivalsAndDepartures().size() > 0 ?
+            response.getEntry().getArrivalsAndDepartures().get(0)
+                    .getRouteId().split("_")[0] :
+            "";
+
+    response.getEntry().getArrivalsAndDepartures()
+            .sort((a,b) ->
+                    customRouteSort
+                            .compareRoutes(
+                                    a.getRouteShortName(),
+                                    b.getRouteShortName(),
+                                    customRouteSort,
+                                    agencyId)
+            );
+
+    return response;
   }
 
   public EntryWithReferencesBean<StopsWithArrivalsAndDeparturesV2Bean> getResponse(
           StopsWithArrivalsAndDeparturesBean result) {
-      return entry(getStopsWithArrivalAndDepartures(result));
+
+    EntryWithReferencesBean<StopsWithArrivalsAndDeparturesV2Bean> response = entry(getStopsWithArrivalAndDepartures(result));
+
+    String agencyId = response
+            .getEntry()
+            .getArrivalsAndDepartures().size() > 0 ?
+            response.getEntry().getArrivalsAndDepartures().get(0)
+                    .getRouteId().split("_")[0] :
```

Comment:
Try not to assume the agencyAndId separator will be "_".  Use AgencyAndIdLibrary to convert